src/shortestpaths/desopo-pape.jl

```diff
@@ -0,0 +1,46 @@
+"""
+    struct DEposoPapeState{T, U}
+
+An [`AbstractPathState`](@ref) designed for D`Esopo-Pape shortest-path calculations.
+"""
+struct DEsopoPapeState{T <:Real, U <: Integer} <: AbstractPathState
+    parents::Vector{U}
+    dists::Vector{T}
+end
+
+function DEsopoPape_shortest_path(g::AbstractGraph, 
+    src::U,
```

Comment:
Don't take the eltype from the source vertex, but take it from the graph `g` and then convert `src` to that eltype at the beginning of the functions, so that we can avoid time consuming convertions inside the loops.

Reply:
Ok!

Reply:
>take it from the graph `g` 

As in?

Reply:
`edgetype` of `g`?

Reply:
You either write `g::AbstractGraph{U}` in the signature or you use `U = eltype(g)` later in the code. In Both cases you should write `src::Integer`.

Reply:
No, `eltype` which for graphs is actually the type of the vertex. For other collections, eltype is usual the type of the stored element.

Reply:
cool, i used `eltype(g)`